pkg/virt-operator/resource/apply/reconcile.go

```diff
@@ -136,11 +136,53 @@ func InjectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev
 	if podSpec == nil {
 		podSpec = &corev1.PodSpec{}
 	}
+
 	if componentConfig == nil || componentConfig.NodePlacement == nil {
-		componentConfig = &v1.ComponentConfig{
-			NodePlacement: &v1.NodePlacement{},
+		switch nodePlacementOption {
+		case AnyNode:
+			componentConfig = &v1.ComponentConfig{NodePlacement: &v1.NodePlacement{}}
+
+		case RequireControlPlanePreferNonWorker:
```

Comment:
Please consider exporting the building of `ComponentConfig` and / or its complex-objects to function (e.g: newPlacementRuleAnyNode..) its not easy to follow already, lets not add more to it :slightly_smiling_face: